lib/Fregot/Builtins/Basics.hs

```diff
@@ -421,6 +422,11 @@ builtin_trim_space = Builtin
     (Ty.string 🡒 Ty.out Ty.string) $ pure $
     \(Cons str Nil) -> return $! T.dropAround isSpace str
 
+builtin_type_name :: Monad m => Builtin m
+builtin_type_name = Builtin
+    (Ty.any 🡒 Ty.out Ty.string) $ pure $
+    \(Cons v Nil) -> return $! T.pack $ describeValue v
```

Comment:
If we didn't use pack here, we'd run into trouble with `ToVal Char` (from `ToVal a => ToVal [a]`)...